### DISCUSSION TITLE
How to navigate to Different PaymentView in POS based on user payment method selection

### DISCUSSION THREAD

**Discussion by LavanyaNiki** [microsoft/Dynamics365Commerce.Solutions#189] [Q&A] [answered]
In the payment button grid at POS level, there may be custom buttons related to different custom Operation Ids. 
This value (Operation Id) would correspond to a group of payment methods the store has. 
For example: Operation Id of one of these buttons would be related to this group of payment methods (Omnichannel Gift Card, Virtual Gift Card, Return Gift Card, etc) which would be filtered.

![image](https://github.com/microsoft/Dynamics365Commerce.Solutions/assets/140384016/d8454f8e-28f4-4b84-b5c5-debeb3294810)

The customized POS operation will require filtering payment methods to present the payment methods related to that custom button and Operation Id.

<img width="306" alt="image" src="https://github.com/microsoft/Dynamics365Commerce.Solutions/assets/140384016/e606cc23-d4c3-4fe6-8310-d78e92240f48">

![image](https://github.com/microsoft/Dynamics365Commerce.Solutions/assets/140384016/2f546b63-dc46-49ec-b0af-55a512e82b1b)

Operation Id will be used to filter the payment methods of the store, so that only a few payment methods will appear in the payment window when the user clicks on custom button.
![image](https://github.com/microsoft/Dynamics365Commerce.Solutions/assets/140384016/5b00e230-5bce-4f87-9e6a-7a61a55e5b50)

Based on POS operation ID we can filter the payment methods when the user clicks on custom button.  This is working , we can filter the payment methods and show them.
However, these options are just normal filter options and these can’t operate as normal payment methods.
We have tried to apply navigation to different payment method screens based on user selection, but current component doesn’t directly support the Payment view navigation.
![image](https://github.com/microsoft/Dynamics365Commerce.Solutions/assets/140384016/93c3c3ab-2a63-426e-b04d-2532ac03dcce)

We want to have different payment view navigations based on selected payment method by the user.
With POS extension is it possible to navigate from one operation screen to different payment view based on user payment selection. How to achieve this?

**Comment by madyke** [comment, marked as answer]
Hello @LavanyaNiki - unfortunately, we don't support calling the payment operations/navigating to the payment view from extensions today so it would not be possible to implement the requirement exactly as requested. Our recommendation to achieve a similar result to meet the business requirement would be:
1. Configure the `ConcludeTransaction` operation as a single point of entry on the button grid.
2. Implement a `PreSelectTransactionPaymentMethodTrigger` that will show a `ListInputDialog` where the user will select which payment method group to use. The trigger implementation will then filter out the payment methods that are not applicable to the scenario, and the payment workflow will continue using the standard payment code.

**Comment by LavanyaNiki** [comment]
Hello @madyke: Thank you for the input. 
Point no 2 is more suitable solution for us.
